sdk/core/core/src/_az_span.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef __az_SPAN_H
```

Comment:
We can't use `__`. it's reserved. I would suggest to have `src/az_span_private.h` name. The only exceptions is `inc/_az_cfg_...`.